pkg/api/handlers/brokers_handler.go

```diff
@@ -46,3 +52,36 @@ func (bh *BrokerHandler) HandleGet() rest.Handler {
 	}
 	return rest.Handler(handler)
 }
+
+// KafkaCreateHandler is the function that processes requests at the /brokers/kafka endpoint
+func (bh *BrokerHandler) KafkaCreateHandler() rest.Handler {
+	logger.Info("handling the brokers' kafka route request")
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		// decode into the bytes of yaml file
+		var content models.BrokerConfigDI
+		err := json.NewDecoder(r.Body).Decode(&content)
+		if err != nil {
+			rest.ERROR(w, err)
+			return
+		}
+
+		var kafkaConfig sidecars.KafkaConfig
+		// parsing the bytes into a Kafka config structure
+		err = yaml.Unmarshal(content.FileContents, &kafkaConfig)
+		if err != nil {
+			rest.ERROR(w, err)
+			return
+		}
+
+		if err = bh.Brokers.Create(
+			metabrokers.BrokerStatus(metabrokers.Kafka),
```

Comment:
Shoud this be hard coded for kafka?

Reply:
Yes, as this is the specific handler for Kafka sidecar creation